### DISCUSSION TITLE
`context manager` 適當翻譯之相關討論

### DISCUSSION THREAD

**Discussion by mattwang44** [python/python-docs-zh-tw#400] [General]
## Root cause

`context manager` 為 Python 獨有名詞，自然在台灣教育部相關網站找不到可以參考的英文翻譯，網路上能找的技術文章亦無統一用詞，在 review [#335](https://github.com/python/python-docs-zh-tw/pull/335#discussion_r1003073080) 也遇到困難。

## Survey

- 網路文章方面，台灣大多翻譯為「情境管理器」（所以目前已翻譯文件也多譯為「情境管理器」）、中國大多譯為「上下文管理器」。
- 請教專家 PeterWolf 推薦用詞為「資源管理器」，完整回覆如下
  <details>
  <summary>click to expand</summary>

  ```
  Context 一詞用在語言裡，指的是「合情合理 (尤其是合情)」的解讀。

  例如「你再做一次試試看」在不同的 context 之下，有不同的意思。因此它是「情」境。

  但程式語言不講情的，它是講理，講結構的。因此「上下文、前後文」會是更適合的譯詞。

  然而，現代漢語有強烈的雙音節傾向，因此「上下文/前後文」會破壞這個音韻節奏的要求。所以它也被排除了。

  其它的「文脈、脈絡…」等等都不是常人能懂的詞彙。而「語境」更是讓早早就「文理分家」的台灣人分不清楚，我明明是來寫程式的，怎麼會冒出一個國文課才有的詞彙呢？

  看來，我們得要放棄字面上的翻譯，直接以 context manager 的功能來設計譯文。既然它是一個「自行依使用者的設定而管理資源的可操作與可回收標記的機制」那麼稱為「資源管理器」或是「資源總監」似乎都是不錯的說法。
  ```
  </details>

- 我在 twitter 發起[投票](https://twitter.com/mattwang44/status/1588010804414013440)，大多數人習慣「情境管理器」，但並沒有和習慣「上下文管理器」的人數相差太多。不過觸及率極低，不是很有參考價值。

  <details>
  <summary>click to expand</summary>

  <img width="598" alt="image" src="https://user-images.githubusercontent.com/24987826/209531247-f520acc5-bd69-45df-956a-c447e868caf6.png">
  </details>

## Status Quo

目前譯文仍維持使用「情境管理器」、新的翻譯也先遵循這個慣例，但希望未來能夠收集更多使用者意見並討論出最佳方案

**Comment by xchux** [comment]
依照一些過往紀錄，還有使用「環境管理器」
- [Pycon Taiwan 紀錄分享](http://dokelung.me/category/python/dont-reinvent-sandwich-2/#:~:text=%E5%99%A8%20(decorator)%20%E5%92%8C-,%E7%92%B0%E5%A2%83%E7%AE%A1%E7%90%86%E5%99%A8,-(context%20manager))

**Comment by mattwang44** [comment]
「環境管理器」會不會比較容易跟 pyenv 一類工具的代稱（版本管理）混淆？

**Reply by xchux** [reply]
同意，有混淆風險

**Comment by husohome** [comment]
可不可以翻成「資源情境」管理器 因為這樣真的比較符合 context manager 在 Python 的用途 @@
